amp/WEB-INF/src/org/digijava/module/aim/dbentity/AmpActivityFields.java

```diff
@@ -61,6 +63,7 @@
 @LocationTotalPercentage(groups = API.class)
 @SectorsTotalPercentage(groups = API.class)
 @ProgramTotalPercentage(groups = API.class)
+@InterchangeableValidator(ComponentFundingOrgRoleValidator.class)
```

Comment:
You're right.

But this works only for constraints defined at class level because those are invoked after the whole json was converted to model object. Accessing parent objects in constraints defined on fields would be incorrect because of the way validation is integrated into importer, in such case validator could access something not yet initialized.

Will move to avoid crowding.

Reply:
From what I've seen, the validation workflow is triggered after all conversion happens. Also if a field based validator wants to access the parent, then it shouldn't be a field but a class level validator. `AmpComponentFunding` in this concrete case is a class.

So if there is no reason to keep at activity level, then it would be great that you can move it. Thanks.

Reply:
On a second thought current location doesn't look so bad. Validator will require access to fields at activity level in order to obtain correct organization roles. The APIField from ConstraintValidator#isValid is a temporary trade off, it should have been just type information about the value currently being validated. If we implement APIField#getParent the it will be impossible to get rid off APIField in isValid. Or we can think of other ways of telling the validator the info about fields, but in my opinion this complicates the validator itself. The locality of a validator is a good property, and in that case we need access to both org roles and components in order to validate correctly. What do you think?

Reply:
I see, so beside navigating to the activity through `AmpComponentFunding.component.activity`, the validator needs also the org role APIField. I wonder if we can add the root APIField to `ConstraintValidatorContext`. Btw this will also allow the component validator not be called at all when components are disabled, though validator can have an extra FM option.
However it is not critical and we can give another thought later, also see how things evolve.